core/pkg/api/grpc/rack.go

```
@@ -59,11 +60,21 @@ var (
 )
 
 func translateRackForward(r *api.Rack) *gapi.Rack {
-	return &gapi.Rack{Key: uint32(r.Key), Name: r.Name}
+	gr := &gapi.Rack{Key: uint32(r.Key), Name: r.Name}
+	if r.Status != nil {
+		gr.Status, _ = status.TranslateToPB[rack.StatusDetails](status.Status[rack.StatusDetails](*r.Status))
+	}
+	return gr
 }
 
 func translateRackBackward(r *gapi.Rack) *api.Rack {
-	return &api.Rack{Key: rack.Key(r.Key), Name: r.Name}
+	ar := &api.Rack{Key: rack.Key(r.Key), Name: r.Name}
+	if r.Status != nil {
+		s, _ := status.TranslateFromPB[rack.StatusDetails](r.Status)
```

shouldn't ignore error